src/skin/tooltipupdater.cpp

```diff
@@ -0,0 +1,170 @@
+#include "widget/wslidercomposed.h"
+#include "widget/wpushbutton.h"
+#include "tooltipupdater.h"
+
+TooltipShortcutUpdater::TooltipShortcutUpdater() { }
+
+TooltipShortcutUpdater::~TooltipShortcutUpdater() {
+    // TODO(Tomasito) Delete WidgetTooltipWatchers
+}
+
+void TooltipShortcutUpdater::addWatcher(QList<ConfigKey> configKeys, WBaseWidget* pWidget) {
+    WSliderComposed* sliderComposed = toSliderComposed(pWidget);
```

Comment:
You could get rid of these up-casts, by a WBaseWidget::createTooltipWatcher() function.